tiledb/common/memory_tracker.h

```diff
@@ -122,7 +122,8 @@ enum class MemoryType {
   METADATA,
   DOMAINS,
   TILE_SUMS,
-  TILE_WRITER_DATA
+  TILE_WRITER_DATA,
```

Comment:
Matter of taste certainly, but how about naming this `WRITER_TILE_DATA` to make it clearer that it's for "data" that belong to `WriterTile`s and not to "writer_data" that belong to `Tile`s ?

Reply:
Done